tracer/src/Datadog.Trace.Tools.Runner/Checks/AgentConnectivityCheck.cs

```diff
@@ -0,0 +1,84 @@
+// <copyright file="AgentConnectivityCheck.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Spectre.Console;
+
+namespace Datadog.Trace.Tools.Runner.Checks
+{
+    internal class AgentConnectivityCheck
+    {
+        public static Task<bool> Run(ProcessInfo process)
+        {
+            // Extract the agent information from the environment variables
+            string url;
+
+            if (!process.EnvironmentVariables.TryGetValue(ConfigurationKeys.AgentUri, out url))
+            {
+                process.EnvironmentVariables.TryGetValue(ConfigurationKeys.AgentHost, out var rawHost);
+                process.EnvironmentVariables.TryGetValue(ConfigurationKeys.AgentPort, out var rawPort);
+
+                var host = rawHost ?? ExporterSettings.DefaultAgentHost;
+
+                int port;
+
+                if (!int.TryParse(rawPort, out port))
+                {
+                    port = ExporterSettings.DefaultAgentPort;
+                }
+
+                url = $"http://{host}:{port}";
+            }
+
+            url ??= $"http://{ExporterSettings.DefaultAgentHost}:{ExporterSettings.DefaultAgentPort}";
+
+            AnsiConsole.WriteLine($"Detected agent url: {url}. Note: this url may be incorrect if you configured the application through a configuration file.");
```

Comment:
😂 Yeah, I should have read one line further apparently. I wonder what proportion of users this would affect